src/tribler/core/upgrade/tribler_db/migration_chain.py

```diff
@@ -0,0 +1,53 @@
+import logging
+from typing import Callable, Iterator, List, Optional
+
+from tribler.core.components.database.db.tribler_database import TriblerDatabase
+from tribler.core.upgrade.tribler_db.decorator import has_migration_metadata
+from tribler.core.upgrade.tribler_db.scheme_migrations.scheme_migration_0 import scheme_migration_0
+from tribler.core.utilities.path_util import Path
+from tribler.core.utilities.simpledefs import STATEDIR_DB_DIR
+
+
+class TriblerDatabaseMigrationChain:
+    """ A chain of migrations that can be executed on a TriblerDatabase.
+
+    To create a new migration, create a new function and decorate it with the `migration` decorator. Then add it to
+    the `DEFAULT_CHAIN` list.
+    """
+
+    DEFAULT_CHAIN = [
+        scheme_migration_0,
+        # add your migration here
+    ]
+
+    def __init__(self, state_dir: Path, chain: Optional[List[Callable]] = None):
+        self.logger = logging.getLogger(self.__class__.__name__)
+        self.state_dir = state_dir
+
+        db_path = self.state_dir / STATEDIR_DB_DIR / 'tribler.db'
+        self.logger.info(f'Tribler DB path: {db_path}')
+        self.db = TriblerDatabase(str(db_path), check_tables=False) if db_path.is_file() else None
+
+        self.migrations = chain or self.DEFAULT_CHAIN
+
+    def execute(self) -> bool:
+        """ Execute all migrations in the chain.
+
+        Returns: True if all migrations were executed successfully, False otherwise.
+        An exception in any of the migrations will halt the execution chain and be re-raised.
+        """
+
+        if not self.db:
+            return False
+
+        for _ in self.steps():
+            ...
+
+        return True
+
+    def steps(self) -> Iterator:
```

Comment:
Regarding the terminology: I'd say we have a migration chain that consists of migrations, and a single migration can possibly consist of several steps. Here, in the `for` loop, we iterate over migrations and not over the steps of an individual migration. So, it is better to rename it from `steps` to `migrations` to avoid confusion.

Reply:
We already have a property named `migrations` that stores all the migrations for this chain.

`def steps(self)` represents the execution steps for the migration chain.

Reply:
I mean, it is pretty common to say that an individual migration may consist of several steps. For that reason, using the word "steps" when referring to migrations looks confusing. Maybe we can rename `steps` to something like `find_suitable_migrations(current_schema_version)`.